ee/codegen/src/__test__/__snapshots__/project.test.ts.snap

```
@@ -1,5 +1,133 @@
+from vellum.workflows.nodes.displayable import InlinePromptNode as BaseInlinePromptNode
+from vellum.workflows.types.core import MergeBehavior
+
+
+def my_test_function(arg1: str, arg2: str) -> str:
+    """Processes input data and returns formatted output"""
+    return f"arg1: {arg1}, arg2: {arg2}"
+
+
+class InlinePromptNode(BaseInlinePromptNode):
+    ml_model = "gpt-4"
+    blocks = [
+        JinjaPromptBlock(template="""Hello world"""),
+    ]
+    functions = [
+        {
+            "type": "CODE_EXECUTION",
+            "name": "my_test_function",
+            "description": "Some sample test function",
+            "src": 'def my_test_function(arg1: str, arg2: str) -> str:\\n    """Processes input data and returns formatted output"""\\n    return f"arg1: {arg1}, arg2: {arg2}"',
+            "definition": {
+                "name": "my_test_function",
+                "description": "Some sample test function",
+                "parameters": {
+                    "type": "object",
+                    "required": [
+                        "arg1",
+                        "arg2",
+                    ],
+                    "properties": {
+                        "arg1": {
+                            "type": "string",
+                        },
+                        "arg2": {
+                            "type": "string",
+                        },
+                    },
+                },
+                "state": None,
+                "cache_config": None,
+                "forced": None,
+                "strict": None,
+            },
+        },
+    ]
+    parameters = PromptParameters(
+        stop=[],
+        temperature=0.7,
+        max_tokens=None,
+        top_p=None,
+        top_k=None,
+        frequency_penalty=None,
+        presence_penalty=None,
+        logit_bias=None,
+        custom_parameters=None,
+    )
+
+    class NodeTrigger(BaseInlinePromptNode.Trigger):
+        merge_behavior = MergeBehavior.AWAIT_ATTRIBUTES
```

[nit] for codegen tests, we should try to zero in and just generate the piece of logic that we care about. If this is mostly because of the factory, then we could [oos] updates to the factory such that we codegen less unrelated logic overall

---

Agreed. This is related to the factory and definitely would love to start incrementally refactoring it to a true builder pattern where just tack on what we want and not have it default to random stuff